### PR TITLE
added: session_name, maximum_n_backups

### PR DESCRIPTION
**Pull request by aidenlt**

### CODE REVIEW

File: models/ModelBase.py [hunk 1]
```diff
@@ -192,7 +194,7 @@ def __init__(self, is_training=False,
         self.preview_history_writer = None
         if self.is_training:
             self.preview_history_path = self.saved_models_path / ( f'{self.get_model_name()}_history' )
-            self.autobackups_path     = self.saved_models_path / ( f'{self.get_model_name()}_autobackups' )
+            self.autobackups_path     = self.saved_models_path / ( f'{self.get_model_name()}__autobackups' )
```
Comment (jmhummel) [inline models/ModelBase.py:197]: Not sure if this was intentional, but you changed the suffix `_autobackups` to `__autobackups` (two underscores)

File: models/ModelBase.py [hunk 2]
```diff
@@ -276,10 +278,18 @@ def load_or_def_option(self, name, def_value):
     def ask_override(self):
         return self.is_training and self.iter != 0 and io.input_in_time ("Press enter in 2 seconds to override model settings.", 5 if io.is_colab() else 2 )
 
+    def ask_session_name(self, default_value=""):
+        default_session_name = self.options['session_name'] = self.load_or_def_option('session_name', default_value)
+        self.options['session_name'] = io.input_str("Session name", default_session_name, help_message="String to refer back to in summary.txt and in autobackup foldername")
+
     def ask_autobackup_hour(self, default_value=0):
         default_autobackup_hour = self.options['autobackup_hour'] = self.load_or_def_option('autobackup_hour', default_value)
         self.options['autobackup_hour'] = io.input_int(f"Autobackup every N hour", default_autobackup_hour, add_info="0..24", help_message="Autobackup model files with preview every N hour. Latest backup located in model/<>_autobackups/01")
```
Comment (jmhummel) [inline models/ModelBase.py:287]: We should probably update the help message here. Latest backup is no longer "01", it's the last folder when sorted name ascending (or first when when sorted name descending.

File: models/ModelBase.py (hunk 1 above)
Comment (aidenlt) [reply models/ModelBase.py:197]: I don't recall ever changing that, must've been accidental. Will be resolved next commit 👍

File: models/ModelBase.py [hunk 3]
```diff
@@ -405,33 +415,30 @@ def create_backup(self):
...
-                previews = self.get_previews()
-                plist = []
-                for i in range(len(previews)):
-                    name, bgr = previews[i]
-                    plist += [ (bgr, idx_backup_path / ( ('preview_%s.jpg') % (name))  )  ]
-
-                if len(plist) != 0:
-                    self.get_preview_history_writer().post(plist, self.loss_history, self.iter)
+        # Create new backup
+        idx_str = datetime.datetime.now().strftime('%Y%m%dT%H%M%S') + "_" + self.session_name
```
Comment (jmhummel) [inline models/ModelBase.py:419]: I think we should add a line before this to build the suffix (and only include the `_` if it's non-empty.
e.g.:
```python
session_suffix = f'_{self.session_name}' if self.session_name else ''
idx_str = datetime.datetime.now().strftime('%Y%m%dT%H%M%S') + session_suffix
```

File: models/ModelBase.py [hunk 4]
```diff
@@ -405,33 +415,30 @@ def create_backup(self):
...
+        idx_backup_path = self.autobackups_path / idx_str
+        idx_backup_path.mkdir()
+        for filename in bckp_filename_list:
+            shutil.copy(str(filename), str(idx_backup_path / Path(filename).name))
+            previews = self.get_previews()
```
Comment (jmhummel) [inline models/ModelBase.py:424]: indentation issue - should be outside of for loop, remove a tab

**Review by jmhummel — CHANGES_REQUESTED** [review]
Couple issues to fix, we can merge once they are updated

**Review by jmhummel — APPROVED** [review]

**Pull request closed without merging**